src/uwtools/tests/drivers/test_jedi.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=missing-function-docstring,protected-access,redefined-outer-name
+"""
+JEDI driver tests.
+"""
+import datetime as dt
+import logging
+from pathlib import Path
+from unittest.mock import DEFAULT as D
+from unittest.mock import Mock, call, patch
+
+import yaml
+from iotaa import asset, external
+from pytest import fixture
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers import jedi
+from uwtools.scheduler import Slurm
+from uwtools.tests.support import regex_logged
+
+# Fixtures
+
+
+@fixture
+def cycle():
+    return dt.datetime(2024, 2, 1, 18)
+
+
+# Driver fixtures
+
+
+@fixture
+def config(tmp_path):
+    return {
+        "jedi": {
+            "execution": {
+                "batchargs": {
+                    "export": "NONE",
+                    "nodes": 1,
+                    "stdout": "/path/to/file",
+                    "walltime": "00:02:00",
+                },
+                "envcmds": [
+                    "module load some-module",
+                    "module load jedi-module",
+                ],
+                "executable": "/path/to/qg_forecast.x",
+                "mpiargs": ["--export=ALL", "--ntasks $SLURM_CPUS_ON_NODE"],
+                "mpicmd": "srun",
+            },
+            "configuration_file": {
+                "base_file": str(tmp_path / "jedi.yaml"),
+                "update_values": {"baz": "qux"},
+            },
+            "files_to_copy": {
+                "foo": "/path/to/foo",
+                "bar/baz": "/path/to/baz",
+            },
+            "files_to_link": {
+                "foo": "/path/to/foo",
+                "bar/baz": "/path/to/baz",
+            },
```

Comment:
For testing purposes, it's ok if the `files_to_link` and `files_to_copy` blocks appear to deal with the same files, but if this gets copied to the `docs/shared/jedi.yaml` file to flesh that out, it would probably be a better example for users if the filenames differ so that it doesn't look like we're linking and copying the same files to the run directory.